pallets/providers/src/lib.rs

```diff
@@ -447,35 +456,61 @@ pub mod pallet {
-    /// marking them as insolvent. If a provider is marked as insolvent, the network (e.g users, other providers) can issue
-    /// `add_redundancy` requests to replicate the data loss if it was a BSP. If it was an MSP, the user can decide to move their
-    /// buckets to another MSP or delete their buckets (as they normally can).
+    /// Processed in the `on_idle` hook.
     ///
-    /// The relay chain block is used to ensure we have a predictable way to determine how much time we allocate to the provider to
-    /// top up their deposit.
+    /// Provider top up expiration items are ignored and cleared if the provider is not found in the [`AwaitingTopUpFromProviders`] storage.
+    /// Providers are removed from `AwaitingTopUpFromProviders` storage when they have successfully topped up their deposit.
+    /// If they are still part of the `AwaitingTopUpFromProviders` storage after the expiration period, they are marked as insolvent.
     #[pallet::storage]
-    pub type GracePeriodToSlashedProviders<T: Config> = StorageDoubleMap<
+    pub type ProviderTopUpExpirations<T: Config> = StorageMap<
         _,
         Blake2_128Concat,
-        RelayChainBlockNumber,
-        Blake2_128Concat,
-        ProviderIdFor<T>,
-        (),
+        RelayBlockNumber<T>,
+        BoundedVec<ProviderIdFor<T>, T::MaxExpiredItemsInBlock>,
+        ValueQuery,
     >;
 
-    /// Storage providers currently awaited for to top up their deposit. This storage holds the current amount that the provider was
-    /// slashed for.
+    /// A pointer to the earliest available block to insert a new storage request expiration.
     ///
-    /// This is primarily used to lookup providers, restrict certain operations while they are in this state.
+    /// This should always be greater or equal than current block + [`Config::StorageRequestTtl`].
     #[pallet::storage]
-    pub type AwaitingTopUpFromProviders<T: Config> =
-        StorageMap<_, Blake2_128Concat, ProviderIdFor<T>, TopUpMetadata>;
+    pub type NextAvailableProviderTopUpExpirationBlock<T: Config> =
+        StorageValue<_, RelayBlockNumber<T>, ValueQuery>;
+
+    /// A pointer to the starting block to clean up expired storage requests.
+    ///
+    /// If this block is behind the current block number, the cleanup algorithm in `on_idle` will
+    /// attempt to accelerate this block pointer as close to or up to the current block number. This
+    /// will execute provided that there is enough remaining weight to do so.
+    #[pallet::storage]
+    pub type NextStartingBlockToCleanUp<T: Config> = StorageValue<_, BlockNumberFor<T>, ValueQuery>;
+
+    /// A map of insolvent providers who have failed to top up their deposit before the end of the expiration.
+    ///
+    /// Providers are marked insolvent by the `on_idle` hook.
+    ///
+    /// This stores the block number at which the provider was marked insolvent.
```

Comment:
yes good point

Reply:
this is using the payment streams tick so that the payment streams implementation can charge the user based on the time that the provider was solvent forc